mapper-runtime/src/main/java/com/datastax/oss/driver/api/mapper/annotations/Transient.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.mapper.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotates the field or getter of an {@link Entity} property, to indicate that it will not be
+ * mapped to any column (neither during reads nor writes).
+ *
+ * <p>Example:
+ *
+ * <pre>
+ * &#64;Transient private int notAColumn;
+ * </pre>
+ *
+ * This information is used by the mapper processor to exclude the property from generated queries.
+ *
+ * <p>Please note that {@code Transient} takes precedence over {@link PartitionKey} and {@link
+ * ClusteringColumn} annotations.
+ */
+@Target({ElementType.FIELD, ElementType.METHOD})
+@Retention(RetentionPolicy.CLASS)
+public @interface Transient {}
```

Comment:
Some things that the 3.x mapper has that this (currently) doesn't:

1. There was a way of specifying a [`PropertyTransienceStrategy`](https://github.com/datastax/java-driver/blob/3.x/driver-mapping/src/main/java/com/datastax/driver/mapping/PropertyTransienceStrategy.java), which allowed one to choose whether properties need to be explicitly annotated or not to be included.  Since we don't have a `@Column` annotation, this didn't seem appropriate here.

2. You could provide a global configuration for specifying transient property names, i.e.: [`addTransientPropertyNames`](https://github.com/datastax/java-driver/blob/3.x/driver-mapping/src/main/java/com/datastax/driver/mapping/DefaultPropertyMapper.java#L179-L242).  I think a good solution for this might be to add an entity-scoped annotation named `@TransientProperties` that is akin to jackson-annotations' [`JsonIgnoreProperties`](https://fasterxml.github.io/jackson-annotations/javadoc/2.6/com/fasterxml/jackson/annotation/JsonIgnoreProperties.html), i.e.:

```java
@TransientProperties({"a", "b", "c", "d"})
@Entity
class MyEntity {}
```

what do you think?

Reply:
1. I guess the question is: do we need an `@Column` annotation? I'm not really convinced, mapping all by default sounds pretty good as long as there's a proper exclusion mechanism.
Also this can always be added later as long as it defaults to "map all", so no need to rush it.

2. That sound like a good idea, for those cases where you inherit from a class that you don't own, and which has properties that you don't want to map.

Reply:
1. Agreed that this isn't compelling enough to add a `@Column`.
2. :+1:, I'll add that.